Source/core/Time.cpp

```diff
@@ -968,21 +972,25 @@ namespace Core {
     {
         struct tm source{};
 
-        source.tm_year = year - 1900;
-        source.tm_mon = month - 1;
-        source.tm_mday = day;
-        source.tm_hour = hour;
-        source.tm_min = minute;
-        source.tm_sec = second;
-        source.tm_isdst = -1; // make sure dst is calculated automatically
+        if ((year >= 1900) && ((month >= 1) && (month <= 12)) && ((day >= 1 ) && (day <= 31)) &&
```

Comment:
And perhaps better not to copy these kind of algorithms but to make a helper for them so we only have the code once (could be of use later on somewhere else anyway)

Reply:
Sure, will use a method to handle this

Reply:
Added method to handle this and called from both Windows and posix version